### PR TITLE
imgcodecs: Add rgb flag for imread and imdecode

### PR DESCRIPTION
**Pull request by zihaomu**
Try to `imread` images by RGB to save R-B swapping costs.

## How to use it?
```
img_rgb = cv2.imread("PATH", IMREAD_COLOR_RGB) # OpenCV decode the image by RGB format.
```

## TODO
- [x] Fix the broken code
- [x] Add imread rgb test
- [x] Speed test of rgb mode.

## Performance test

| file name | IMREAD_COLOR  | IMREAD_COLOR_RGB |
| --------- | ------ | --------- |
| jpg01     | 284 ms | 277 ms    |
| jpg02     | 376 ms | 366 ms    |
| png01     | 62 ms  | 60 ms     |
| Png02     | 97 ms  | 94 ms     |

Test with [image_test.zip](https://github.com/user-attachments/files/15982949/image_test.zip)
```.cpp
string img_path = "/Users/mzh/work/data/image_test/png02.png";
int loop = 20;

TickMeter t;

double t0 = 10000;
for (int i = 0; i < loop; i++)
{
    t.reset();
    t.start();
    img_bgr = imread(img_path, IMREAD_COLOR);
    t.stop();

    if (t.getTimeMilli() < t0) t0 = t.getTimeMilli();
}

std::cout<<"bgr time = "<<t0<<std::endl;

t0 = 10000;
for (int i = 0; i < loop; i++)
{
    t.reset();
    t.start();
    img_rgb = imread(img_path, IMREAD_COLOR_RGB);
    t.stop();
    if (t.getTimeMilli() < t0) t0 = t.getTimeMilli();
}
std::cout<<"rgb time = "<<t0<<std::endl;
``` 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/imgcodecs/src/grfmt_jpeg.cpp [hunk 1]
```diff
@@ -514,10 +514,20 @@ bool  JpegDecoder::readData( Mat& img )
...
+                                icvCvt_CMYK2RGB_8u_C4C3R( buffer[0], 0, data, 0, Size(m_width,1) );
+                        }
                         else
-                            icvCvt_CMYK2BGR_8u_C4C3R( buffer[0], 0, data, 0, Size(m_width,1) );
+                        {
+                            if( cinfo->out_color_components == 3 )
+                                icvCvt_RGB2BGR_8u_C3R( buffer[0], 0, data, 0, Size(m_width,1) );
+                            else
+                                icvCvt_CMYK2BGR_8u_C4C3R( buffer[0], 0, data, 0, Size(m_width,1) );
+                        }
```
Comment (asmorkalov) [inline modules/imgcodecs/src/grfmt_jpeg.cpp:530]: If and else branch are the same.

File: modules/imgcodecs/test/test_precomp.hpp [hunk 2]
```diff
@@ -51,6 +51,12 @@ void PrintTo(const ImreadModes& val, std::ostream* os)
         v &= ~IMREAD_IGNORE_ORIENTATION;
         *os << "IMREAD_IGNORE_ORIENTATION" << (v == 0 ? "" : " | ");
     }
+    if ((v & IMREAD_COLOR_RGB) != 0)
+    {
+        CV_Assert(IMREAD_COLOR_RGB == 256);
```
Comment (asmorkalov) [inline modules/imgcodecs/test/test_precomp.hpp:56]: Assert is redundant here

**Comment by asmorkalov** [comment]
I propose to extend existing perf tests with RGB case. Just to be sure that the new mode does not introduce significant degradation.

File: modules/imgcodecs/test/test_precomp.hpp (hunk 2 above)
Comment (zihaomu) [reply modules/imgcodecs/test/test_precomp.hpp]: Done

File: modules/imgcodecs/test/test_png.cpp [hunk 3]
```diff
@@ -83,6 +83,14 @@ TEST(Imgcodecs_Png, read_color_palette_with_alpha)
...
     EXPECT_EQ(img.at<Vec3b>(0, 1), Vec3b(0, 0, 255));
 
+    img = imread(root + "readwrite/color_palette_alpha.png", IMREAD_COLOR_RGB);
+    ASSERT_FALSE(img.empty());
+    ASSERT_TRUE(img.channels() == 3);
+
+    // pixel is red in RGB
+    EXPECT_EQ(img.at<Vec3b>(0, 0), Vec3b(255, 0, 0));
+    EXPECT_EQ(img.at<Vec3b>(0, 1), Vec3b(255, 0, 0));
+
```
Comment (asmorkalov) [inline modules/imgcodecs/test/test_png.cpp:93]: I propose to add test with RGBA output.

File: modules/imgcodecs/test/test_tiff.cpp [hunk 4]
```diff
@@ -1066,6 +1066,7 @@ const int all_modes[] =
     IMREAD_UNCHANGED,
     IMREAD_GRAYSCALE,
     IMREAD_COLOR,
+    IMREAD_COLOR_RGB,
```
[the remark this answers is not in the document]
Comment (zihaomu) [reply modules/imgcodecs/test/test_tiff.cpp:1069]: got it

File: modules/imgcodecs/include/opencv2/imgcodecs.hpp [hunk 5]
```diff
@@ -78,7 +78,8 @@ enum ImreadModes {
        IMREAD_REDUCED_COLOR_4      = 33, //!< If set, always convert image to the 3 channel BGR color image and the image size reduced 1/4.
        IMREAD_REDUCED_GRAYSCALE_8  = 64, //!< If set, always convert image to the single channel grayscale image and the image size reduced 1/8.
        IMREAD_REDUCED_COLOR_8      = 65, //!< If set, always convert image to the 3 channel BGR color image and the image size reduced 1/8.
-       IMREAD_IGNORE_ORIENTATION   = 128 //!< If set, do not rotate the image according to EXIF's orientation flag.
+       IMREAD_IGNORE_ORIENTATION   = 128, //!< If set, do not rotate the image according to EXIF's orientation flag.
+       IMREAD_COLOR_RGB            = 256, //!< If set, always convert image to the 3 channel RGB color image.
```
Comment (asmorkalov) [inline modules/imgcodecs/include/opencv2/imgcodecs.hpp:82]: I propose to introduce `IMREAD_COLOR_BGR` as synonym for `IMREAD_COLOR`, or better `IMREAD_COLOR_BGR` is default and `IMREAD_COLOR` as synonym.

**Comment by asmorkalov** [comment]
Perf results for system wide (Ubuntu 16.04) and own (from 3rdparty) on Jetson-tk1 (ARMv7+NEON):
```
Geometric mean (ms)

  Name of Test   system-1  own-1    own-1   
                                      vs    
                                   system-1 
                                  (x-factor)
Decode::JPEG     129.067  129.898    0.99   
Decode_rgb::JPEG 129.084  129.561    1.00   
Encode::JPEG     165.524  118.029    1.40   
decode::PNG       82.793  68.338     1.21   
decode_rgb::PNG   79.697  63.584     1.25   
encode::PNG      135.514  116.040    1.17 
```

**Review by asmorkalov — APPROVED** [review]
:+1:

**Review by vpisarev — APPROVED** [review]

**Pull request merged**